crates/core/src/node/inner/node_executor.rs

```diff
@@ -0,0 +1,449 @@
+    /// timestamp validity to be confirmed. Block might still not be produced by then.
+    pub async fn enforce_next_timestamp_sync(&self, timestamp: u64) -> anyhow::Result<()> {
+        let (response_sender, response_receiver) = oneshot::channel();
+        self.command_sender
+            .send(Command::EnforceNextTimestamp(timestamp, response_sender))
+            .await
+            .map_err(|_| {
+                anyhow::anyhow!("failed to enforce next timestamp as node executor is dropped")
+            })?;
+        match response_receiver.await {
+            Ok(result) => result,
+            Err(_) => {
+                anyhow::bail!("failed to enforce next timestamp as node executor is dropped")
+            }
+        }
+    }
+
+    /// Request [`NodeExecutor`] to set current timestamp (in seconds). Waits for the
+    /// change to take place.
+    pub async fn set_current_timestamp_sync(&self, timestamp: u64) -> anyhow::Result<i128> {
+        let (response_sender, response_receiver) = oneshot::channel();
+        self.command_sender
+            .send(Command::SetCurrentTimestamp(timestamp, response_sender))
+            .await
+            .map_err(|_| {
+                anyhow::anyhow!("failed to set current timestamp as node executor is dropped")
+            })?;
+
+        match response_receiver.await {
+            Ok(result) => Ok(result),
+            Err(_) => anyhow::bail!("failed to set current timestamp as node executor is dropped"),
+        }
+    }
+
+    /// Request [`NodeExecutor`] to set block timestamp interval (in seconds). Does not wait for the
+    /// change to take place.
+    pub async fn set_block_timestamp_interval(
+        &self,
+        seconds: u64,
+    ) -> Result<(), mpsc::error::SendError<Command>> {
+        self.command_sender
+            .send(Command::SetTimestampInterval(seconds))
+            .await
+    }
+
+    /// Request [`NodeExecutor`] to remove block timestamp interval. Waits for the change to take
+    /// place. Returns `true` if an existing interval was removed, `false` otherwise.
+    pub async fn remove_block_timestamp_interval_sync(&self) -> anyhow::Result<bool> {
+        let (response_sender, response_receiver) = oneshot::channel();
+        self.command_sender
+            .send(Command::RemoveTimestampInterval(response_sender))
+            .await
+            .map_err(|_| {
+                anyhow::anyhow!("failed to remove block interval as node executor is dropped")
+            })?;
+
+        match response_receiver.await {
+            Ok(result) => Ok(result),
+            Err(_) => anyhow::bail!("failed to remove block interval as node executor is dropped"),
+        }
+    }
+}
+
+#[cfg(test)]
+impl NodeExecutorHandle {
+    pub fn test() -> (Self, mpsc::Receiver<Command>) {
+        let (command_sender, command_receiver) = mpsc::channel(128);
+        (NodeExecutorHandle { command_sender }, command_receiver)
+    }
+}
+
+#[derive(Debug)]
+pub enum Command {
```

Comment:
Does it have to be public?

Reply:
Good catch, just needed a small refactoring to make it private